### ISSUE TITLE
Problem with keyboard event simulating

### ISSUE THREAD

**Issue by riophae** [eddyerburgh/avoriaz#102]
Though it's possible to simulate some specific keys to test keyboard events, it's actually too limited - only a few keys are allowed:

https://github.com/eddyerburgh/avoriaz/blob/master/src/Wrapper.js#L373

So I think it's necessary to make it possible to test **any** keys.

Also, as [this article](https://css-tricks.com/snippets/javascript/javascript-keycodes/) suggests, it's best to recognize key press using the `event.which` property rather than `event.keyCode`. But avoriaz has only set value for `keyCode` property. If only `event.which` has been used by the application logic, then we can not test that at all.

In general, the synthetic event should be able to be freely modified to suit any specific needs:

```js
wrapper.trigger('keyup', { which: 8 }) // backspace key
```

**Comment by eddyerburgh** [comment]
I think this is a great suggestion. Are you able to implement it? We could release, along with your other changes, in 4.0.0

**Comment by riophae** [comment]
Yeah, no problem!

**Comment by riophae** [comment]
I think we could outsource the work to this module https://github.com/KenanY/create-event ?

**Comment by eddyerburgh** [comment]
Yes, although we can only use it for keyup, keydown click and dblclick
events

**Comment by riophae** [comment]
These days I have spent some time reading the source code of [triggerhappy](https://github.com/Knape/triggerhappy). It provides a lots more features, including touch events testing. I think we can learn something from it and jest and we will have a lot to discuss about.
But for now, I think there is an urgent need for testing keyboard events and we should implement those as soon as possible.

**Comment by mathijswesterhof** [comment]
I don't know if anyone is already working on this issue/enhancement, but can that person also look at the feature of key + shiftKey=true (and altKey) because that is currently also not functioning. in specific this line  `wrapper.find('input')[0].trigger('keyup', { keycode: 53, shiftKey: true }); `(in this case %)

**Comment by riophae** [comment]
@mathijswesterhof Hi, I'm not working on this. Sorry.

**Comment by eddyerburgh** [comment]
This project isn't actively maintained, although I'm happy to accept PRs.

I recommend using Vue Test Utils, which is based on avoriaz—https://github.com/vuejs/vue-test-utils

**Comment by mathijswesterhof** [comment]
good to know, i'll try to migrate then, see if the problem is solvable on vue-test-utils. if not I'll make a pr there

**Comment by jaapio** [comment]
asasddsadsa
